examples/local/vstream_client.go

```diff
@@ -82,17 +72,372 @@ func main() {
+
+			// commit and other events are safe to flush on, since they indicate the end of a transaction.
+			// Otherwise, there's not much to do with these events.
+			case binlogdatapb.VEventType_COMMIT, binlogdatapb.VEventType_OTHER:
+				// only flush when we have an event that guarantees we're not flushing mid-transaction
+				err = flushFunc()
+				if err != nil {
+					return err
+				}
+
+			// DDL events are schema changes, and we might want to handle them differently than data events.
+			// They are safe to flush on, since they indicate the end of a transaction. If you want to
+			// transparently adjust the destination schema based on DDL events, you would do that here.
+			case binlogdatapb.VEventType_DDL:
+				err = flushFunc()
+				if err != nil {
+					return err
+				}
+
+				// alter the destination schema based on the DDL event
+
+			case binlogdatapb.VEventType_COPY_COMPLETED:
+				// TODO: don't flush until the copy is completed? do some sort of cleanup if we haven't received this?
+
+			// we don't need to do anything with these events, but the timestamp from these and other events
+			// is recorded at the top of the loop for heartbeat monitoring
+			case binlogdatapb.VEventType_HEARTBEAT:
+
+			// even if there are no changes to the tables being streamed, we'll still get a begin, vgtid, and commit
+			// event for each transaction. The other two are used for checkpoints, but nothing to do here.
+			case binlogdatapb.VEventType_BEGIN:
+
+			// journal events are sent on resharding. Unless you are manually targeting shards, vstream should
+			// transparently handle resharding for you, so you shouldn't need to do anything with these events.
+			// You might want to log them for debugging purposes, or to alert on resharding events in case
+			// something goes wrong. After resharding, if the pre-reshard vgtid is no longer valid, you may need
+			// to restart the stream from the beginning.
+			case binlogdatapb.VEventType_JOURNAL:
+
+			// there aren't strong cases for handling these events, but you might want to log them for debugging
+			case binlogdatapb.VEventType_VERSION, binlogdatapb.VEventType_LASTPK, binlogdatapb.VEventType_SAVEPOINT:
+			}
+		}
+	}
+}
+
+// processCustomerRowEvent builds a customer from a row event. It will only process the last row in the event.
+// If you wanted to process all rows in the event, you would need to change the function signature to return a slice.
+func processCustomerRowEvent(fields []*querypb.Field, rowEvent *binlogdatapb.RowEvent) (*Customer, error) {
+	if fields == nil {
+		return nil, errors.New("internal error: unexpected rows without fields")
+	}
+
+	var customer *Customer
+	var err error
+
+	// TODO: I'm not exactly sure how to handle multiple rows in a single event, so I'm just going to take the last one
+	for _, rc := range rowEvent.RowChanges {
+		// ignore deletes
+		if rc.After == nil {
+			continue
+		}
+
+		row := sqltypes.MakeRowTrusted(fields, rc.After)
```

Comment:
`MakeRowTrusted` has a warning that it shouldn't be used except in special circumstances, which require comments about why it is used. Is there a better way to handle this conversion?